kitsune/search/api.py

```diff
@@ -0,0 +1,98 @@
+import itertools
+
+from django.conf import settings
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from kitsune.products.models import Product
+from kitsune.questions.models import QuestionMappingType
+from kitsune.search import es_utils
+from kitsune.sumo.api import GenericAPIException
+from kitsune.wiki.models import DocumentMappingType
+
+
+@api_view(['GET', 'POST'])
+def suggest(request):
+    text = request.body or request.GET.get('q')
+    locale = request.GET.get('locale', settings.WIKI_DEFAULT_LANGUAGE)
+    product = request.GET.get('product')
+    max_questions = request.GET.get('max_questions', 10)
+    max_documents = request.GET.get('max_documents', 10)
+
+    errors = {}
+    try:
+        max_questions = int(max_questions)
+    except ValueError:
+        errors['max_questions'] = 'This field must be an integer.'
+    try:
+        max_documents = int(max_documents)
+    except ValueError:
+        errors['max_documents'] = 'This field must be an integer.'
+    if text is None:
+        errors['q'] = 'This field is required.'
+    if product is not None and not Product.objects.filter(slug=product).exists():
+        errors['product'] = 'Could not find product with slug "{0}".'.format(product)
+    if errors:
+        raise GenericAPIException(400, errors)
+
+    wiki_f = es_utils.F(
+        model='wiki_document',
+        document_category__in=settings.SEARCH_DEFAULT_CATEGORIES,
+        document_locale=locale,
+        document_is_archived=False)
+
+    questions_f = es_utils.F(
+        model='questions_question',
+        question_is_archived=False,
+        question_is_locked=False,
+        question_has_helpful=True)
+
+    if product is not None:
+        wiki_f &= es_utils.F(product=product)
+        questions_f &= es_utils.F(product=product)
+
+    mapping_types = [QuestionMappingType, DocumentMappingType]
+    query_fields = itertools.chain(*[cls.get_query_fields() for cls in mapping_types])
+    query = {}
+    for field in query_fields:
+        for query_type in ['match', 'match_phrase']:
+            key = '{0}__{1}'.format(field, query_type)
+            query[key] = text
```

Comment:
One difference between this and normal search is that boosts aren't being applied. Just worth noting, not sure if it's needed.